crates/turbo-tasks-fs/src/rope.rs

```diff
@@ -157,26 +181,31 @@ impl RopeBuilder {
     ///
     /// This may be called multiple times without issue.
     pub fn finish(&mut self) {
-        if !self.writable.is_empty() {
+        if let Some(uncommited_static) = self.uncommited_static.take() {
```

Comment:
This works with the invariant of uncommitted_static and writable are exclusive.

Maybe encode this into an enum:

``` rust
enum Uncommitted {
  None,
  Static(&'static [u8])
  Owned(Vec<u8>)
}
```

Reply:
Or without None and `Option<Uncommitted>`

Reply:
Done.